pkg/smokescreen/acl/v1/acl.go

```diff
@@ -170,24 +170,28 @@ func (acl *ACL) Validate() error {
 	return nil
 }
 
-// ValidateDomains takes a slice of domains and verifies they conform to
-// smokescreen's domain glob policy.
+// ValidateDomainGlobs takes a slice of domain globs and verifies they conform to smokescreen's
+// domain glob policy.
 //
-// Domains can only contain a single wildcard prefix
-// Domains cannot be represented as a sole wildcard
-func (acl *ACL) ValidateDomains(domains []string) error {
-	for _, d := range domains {
-		if d == "" {
+// Wildcards are valid only at the beginning of a domain glob, and only a single wildcard per glob
+// pattern is allowed. Globs must include text after a wildcard.
+func (acl *ACL) ValidateDomainGlobs(svc string, globs []string) error {
+	for _, glob := range globs {
+		if glob == "" {
 			return fmt.Errorf("glob cannot be empty")
 		}
 
-		if !strings.HasPrefix(d, "*.") && strings.HasPrefix(d, "*") {
-			return fmt.Errorf("%v: domain glob must represent a full prefix (sub)domain", d)
+		if glob == "*" || glob == "*." {
```

Comment:
This is appropriate because anything which wants this behavior should just change the default enforcement action to `open`.